### DISCUSSION TITLE
MusicBee support?

### DISCUSSION THREAD

**Discussion by ghost** [jojo2357/Music-Discord-Rich-Presence#20] [Ideas]
Hello, let me first say that your work is amazing and that you have my gratitude!

I've recently changed music players and I'm now using MusicBee. Would it be possible to add support? Or maybe add it to an [already existing MusicBee plugin](https://github.com/Harmon758/mb_DiscordRichPresence)? (I also may have [mentioned your repo](https://github.com/Harmon758/mb_DiscordRichPresence/issues/6) without letting you know, sorry about that. I'm still getting used to this github stuff.) 

I would like to see MusicBee support to be added (if you're okay with it, of course) and thanks again!

**Comment by jojo2357** [comment]
That is totally doable (i think). I'll look into it and let you know.

And thanks for your kind words. This is why I do open source dev :D

**Reply by jojo2357** [reply]
Okay @Javonski let me know if the latest release works for you

**Reply by jojo2357** [reply]
![image](https://user-images.githubusercontent.com/66704796/111106023-822f2c80-8511-11eb-9b01-c434f94f469f.png)
![image](https://user-images.githubusercontent.com/66704796/111106054-93783900-8511-11eb-9b38-8fc6e1a850e7.png)
:)

**Reply by jojo2357** [reply]
lol probably shouldve taken those in a different context whatever lmao

**Reply by ghost** [reply]
Woah bruh! That was really fast!
![image](https://user-images.githubusercontent.com/66522068/111109218-4b025f00-8595-11eb-9808-344a907a5e6c.png)
It works perfectly, thank you so much, this is so amazing!

**Reply by jojo2357** [reply]
oh and i would recommend that you add a music bee icon with key `musicbee_small` so it shows up in the bottom right there like in mine

**Reply by ghost** [reply]
Ohh so that's what it is, I see. Thank you again!

**Reply by ghost** [reply]
Hello again! So, it works wonderfully! The only problems I've discovered so far is that it doesn't open automatically for MusicBee and I need to launch Groove first. Another is that when closing MusicBee, it doesn't close the Rich Presence on Discord and it stays on the last song played. I'm guessing it needs to be linked to the MusicBee app as well? If so, how do I do that? Do I just change the path inside "GroovyRP\bin\Release\filelocation.dat" to the path to MusicBee.exe?

Also, a less important question, is it possible to change the tooltip of the small image?

**Reply by jojo2357** [reply]
You dont need to launch groove at all you can use one of the other shortcuts provided

Funny thing is i was just thinking about making a linker for each supported player. I suppose you could do it that way and just put the path to the MusicBee exe for now. 

I can change the small image tooltip, what did you have in mind for it?

**Reply by ghost** [reply]
For the tool tip I was just thinking of something shorter like "MusicBee" or "Playing MusicBee" but it's not really that important lmao.

As for changing the file location, I'm not sure if I'm doing it right. I just need to change the location in "filelocation.dat" and "filelocation.dat" and run "LinkWithGroove.bat", is that correct?

**Reply by jojo2357** [reply]
I believe so. Ill get an actual linker out tomorrow when i get the chance

**Reply by ghost** [reply]
I see, I see. Thank you again for taking the time to work on this!

**Reply by ghost** [reply]
I just now realized that the small image disappears when the song is paused. That's pretty cool! So that's why the image tool tip is what it is.

**Reply by jojo2357** [reply]
If you include an icon keyed `paused` then it will show up ;)

**Reply by jojo2357** [reply]
I made the launchers so please try them out and let me know. If they work ill make a release

**Reply by ghost** [reply]
Nice! They work very smoothly so far. Thanks!
